CMSgov/cmcs-eregulations
# EREGCSC-1619 -- Rollup to Vite

Resolves [EREGCSC-1619](https://jiraent.cms.gov/browse/EREGCSC-1619)

**Description:**
Update Django-specific front end bundle pipeline to allow using Vuetify etc. This will likely mean migrating from Rollup to Vite.

**This pull request changes:**

- Removes Rollup build pipeline, deleting config file and removing all artifacts and dependencies associated with it.
- Moves former Rollup build pipeline to a Vite build pipeline under the directory `eregs-component-lib`
- Upgrades all Vite instances (`eregs-vite` and `eregs-component-lib`) to `v4`
- Upgrades `Makefile` to use new build pipelines for local development (`make watch` and `make watch-all`) and for deployment
   - `make watch` and `make watch-all` will behave as expected:
      - `make watch` will build and watch only the Django template scripts and styles (so nothing related to the Single Page Apps at `/search` or `/resources`)
      - `make watch-all` will build and watch everything -- Django template scripts and styles, as well as the SPA scripts and styles
- Fixes Storybook issue that was introduced when Node was upgraded to v18

**Steps to manually verify this change:**

1. Pull down repository and make sure everything works as expected on your local machine.
   - `make watch`
   - `make watch-all`
   - `make test`
   - `make storybook`
2. Make sure repository is deploying to experimental deployment without issue and all tests passing.  
3. Poke around and compare experimental deployment to prod site to make sure things match.
